### ISSUE TITLE
[hebao] Filtering transfer/approval in TransferModule not always sufficient

### ISSUE THREAD

**Issue by Brechtpd** [Loopring/protocols#1103]
https://github.com/Loopring/protocols/blob/3f449de7275e0c20d45de4718ed9eb4730295a1e/packages/hebao_v1/contracts/modules/transfers/TransferModule.sol#L104

From [Argent's audit](https://github.com/argentlabs/argent-contracts/blob/develop/audit/G0Group-Argent_Upgrade_Nov2019.pdf):

> Some ERC20 contracts use non-standard functions to transfer tokens, such as
increaseAllowance, this means that merely preventing calls of approve and transfer
functions is insufficient to prevent transfer of all ERC20 tokens. This could possibly allow
users to bypass daily spending limits in the TransferManager contract.

I'm not sure which or how many tokens this applies on, but seems like this could certainly be a problem. The fix is not great but it seems like there isn't really a better solution.

**Comment by dong77** [comment]
I don't think there is an ideal solution for this.Even if transfer/approve may have non-standard logics inside, which will break the quota. But I guess breaking the quota is fine in such cases.

**Comment by dong77** [comment]
Is it possible to allow method calls without the check, then verify the after-effects, the balance and allowance changes, to make sure the quota is respected?

**Comment by Brechtpd** [comment]
It's hard to say without knowing what some ERC20 do.

But doing these checks every time for all calls (as we don't know what calls are to ERC20 contracts and which are not) would be quite expensive.

**Comment by dong77** [comment]
Maybe we can just do it once at the end of the user-initiated action that consumes available daily quota.

**Comment by Brechtpd** [comment]
I think we can basically do the same thing as argent and block all calls to contracts which have non-zero price (tokens with zero price don't have any impact on the quota anyway). So we just check
```
require(controller.priceOracle().tokenPrice(contract, 1e18) == 0, "TOKEN_CONTRACT_CALL");
```
in `QuotaTransfers` but not in `ApprovedTransfers` so users can still do anything they want with enough signatures.

**Comment by dong77** [comment]
Please make the judgement call.

**Comment by Brechtpd** [comment]
Fixed in #1106
